src/services/bunker_cases/midterm_slashing_penalty.py

```diff
@@ -275,25 +281,42 @@ def get_validator_midterm_penalty(
     @staticmethod
     def get_validator_midterm_penalty_electra(
         validator: LidoValidator,
-        bound_slashed_validators: list[Validator],
+        slashings: list[Gwei],
         total_balance: Gwei,
+        midterm_penalty_epoch: EpochNumber,
+        report_ref_epoch: EpochNumber
     ) -> Gwei:
         """
         Calculate midterm penalty for particular validator
         https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#modified-process_slashings
         """
-        # We don't know validators effective balances on the moment of slashing,
-        # so we assume that it was at least `effective_balance`
-        slashings = sum((v.validator.effective_balance for v in bound_slashed_validators), Gwei(0))
-        adjusted_total_slashing_balance = min(slashings * PROPORTIONAL_SLASHING_MULTIPLIER_BELLATRIX, total_balance)
-        effective_balance = validator.validator.effective_balance
-        penalty_per_effective_balance_increment = adjusted_total_slashing_balance // (
-            total_balance // EFFECTIVE_BALANCE_INCREMENT
+        slashings_sum = sum(MidtermSlashingPenalty._cut_slashings(slashings, midterm_penalty_epoch, report_ref_epoch))
+        adjusted_total_slashing_balance = min(
+            slashings_sum * PROPORTIONAL_SLASHING_MULTIPLIER_BELLATRIX,
+            total_balance,
         )
-        effective_balance_increments = effective_balance // EFFECTIVE_BALANCE_INCREMENT
+        effective_balance = validator.validator.effective_balance
+        increment = EFFECTIVE_BALANCE_INCREMENT
+        penalty_per_effective_balance_increment = adjusted_total_slashing_balance // (total_balance // increment)
+        effective_balance_increments = effective_balance // increment
         penalty = penalty_per_effective_balance_increment * effective_balance_increments
         return Gwei(penalty)
 
+    @staticmethod
+    def _cut_slashings(
+        slashings: list[Gwei], midterm_penalty_epoch: EpochNumber, report_ref_epoch: EpochNumber
+    ) -> list[Gwei]:
+        """
+        Filters out slashing values based on epochs within a midterm penalty epoch. Slashings is a ring buffer on epochs.
+        @see https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#modified-slash_validator
+        We want to filter out epochs in the past which will not be relevant at the time of midterm penalty epoch.
+        """
+        obsolete_indexes = {
```

Comment:
Maybe let's add sanity check that 
len(slashings) == EPOCHS_PER_SLASHINGS_VECTOR